docs/core/testing/unit-testing-mstest-runner-exit-codes.md

```diff
@@ -0,0 +1,73 @@
+---
+title: MSTest Test Runner Exit Codes
+description: Learn about MSTest Test Runner exit codes.
+author: nohwnd
+ms.author: jajares
+ms.date: 11/28/2023
+ms.custom: 
+---
+
+# Exit codes
+
+MSTest Runner uses known exit codes to communicate test failure or application error. Exit codes start at 0 and are non-negative:
+
+| Exit&nbsp;Code | Reason |
+|:-----|----------|
+| [0](#0) | Success |
+| [1](#1) | Generic failure. |
+| [2](#2) | At least 1 test failed. |
+| [3](#3) | Test session aborted. |
+| [4](#4) | Invalid platform setup. |
+| [5](#5) | Invalid command line. |
+| [6](#6) | Feature not implemented. |
+| [7](#7) | Test host process did not exit gracefully. |
+| [8](#8) | Zero tests ran. |
+| [9](#9) | Minimum expected tests policy violation. |
+| [10](#10) | Test adapter test session failure. |
+
+To enable verbose logging in order to troubleshoot issue(s), see [troubleshooting](./unit-testing-mstest-runner-intro.md#troubleshooting).
+<!-- Setting special name so we can simply link to the number from here, and from error message that is built into TA source code. -->
+
+## <a name="0"></a>0 - Success
```

Comment:
we are using this to hopefully be able to link directly to the exit code from a shortlink looking like aka.ms/mtest-runner-exit-codes/1